### PR TITLE
nss: add NSS 3.52.1 recipe

### PR DESCRIPTION
**Pull request by kenmays**

### CODE REVIEW

File: dev-libs/nss/patches/nss-3.52.1.patchset [hunk 1]
```diff
@@ -265,5 +265,52 @@ index fb2d8d5..e995b4e 100644
...
+index 0796fe5..1249da2 100644
+--- a/lib/certdb/certdb.c
++++ b/lib/certdb/certdb.c
+@@ -2889,8 +2889,8 @@ void
+ CERT_UnlockCertRefCount(CERTCertificate *cert)
+ {
+     PORT_Assert(certRefCountLock != NULL);
+-    PRStatus prstat = PZ_Unlock(certRefCountLock);
+-    PORT_AssertArg(prstat == PR_SUCCESS);
++/*    PRStatus prstat = PZ_Unlock(certRefCountLock);
```
Comment (waddlesplash) [inline dev-libs/nss/patches/nss-3.52.1.patchset:287]: This is definitively wrong and will break things.

**Comment by waddlesplash** [comment]
This change apparently broke LibreOffice because it disabled the actually working version of NSS.

@kenmays, you really need to check over your PRs more; this is not the first time this has happened.

**Comment by waddlesplash** [comment]
And this does not even build:
```
In file included from ../../../dist/public/nss/seccomon.h:27,
                 from cert.h:18,
                 from certdb.c:12:
certdb.c: In function 'CERT_UnlockCertRefCount':
../../../dist/public/nss/secport.h:127:24: error: implicit declaration of function 'PR_ASSERT_ARG'; did you mean 'PR_ASSERT'? [-Werror=implicit-function-declaration]
 #define PORT_AssertArg PR_ASSERT_ARG
                        ^~~~~~~~~~~~~
certdb.c:2893:5: note: in expansion of macro 'PORT_AssertArg'
     PORT_AssertArg(prstat == PR_SUCCESS);
     ^~~~~~~~~~~~~~
```

@kenmays: did you test this at all?!

**Comment by kenmays** [comment]
The recipe wasn't enabled by me. The PR states the prior need for a more recent NSPR before enabling.

**Comment by waddlesplash** [comment]
The buildmaster builds packages using R1B2, what "kernel issue" are you referring to?

You did not provide any of that information in this PR, and you removed the old recipe in this commit, so it looked like this was a simple version bump. You must be more clear.

**Comment by Begasus** [comment]
Got nss 3.52.1 build with nspr 4.26 (gcc2h), checking a build for libreoffice atm, will report later ...

**Comment by kenmays** [comment]
@Begasus - You can bump to NSS 3.54. This will cover PPC/ARM64 vectoring any crypto fixes as well.

**Comment by Begasus** [comment]
> @Begasus - You can bump to NSS 3.54 (if you want). This will cover PPC/ARM64 vectoring and crypto fixes as well.

@kenmays First I'd like to see that we get things working again, if things are OK I'll try to see for the latest version for nss ... (keeping in mind that depending packages are ok with it)

**Pull request merged**